src/ucs/sys/topo/base/topo.c

```diff
@@ -56,60 +56,52 @@ static ucs_topo_global_ctx_t ucs_topo_global_ctx;
 UCS_LIST_HEAD(ucs_sys_topo_methods_list);
 
 
-ucs_sys_topo_method_t *
-ucs_sys_topo_find_method(const char *name)
-{
-    ucs_sys_topo_method_t *method;
-
-    ucs_list_for_each(method, &ucs_sys_topo_methods_list, list) {
-        if (!strcmp(name, method->name)) {
-            return method;
-        }
-    }
-
-    return NULL;
-}
-
 static ucs_sys_topo_method_t *ucs_sys_topo_get_method()
 {
-    ucs_sys_topo_method_t *method = NULL;
+    static ucs_sys_topo_method_t *method = NULL;
+    ucs_sys_topo_method_t *list_method;
     unsigned i;
 
     for (i = 0;
-         (i < ucs_global_opts.topo_methods.count) && (method == NULL);
+         (i < ucs_global_opts.topo_prio.count) && (method == NULL);
          ++i) {
-        method = ucs_sys_topo_find_method(ucs_global_opts.topo_methods.names[i]);
+
+        ucs_list_for_each(list_method, &ucs_sys_topo_methods_list, list) {
+            if (!strcmp(ucs_global_opts.topo_prio.names[i],
+                        list_method->name)) {
+                method = list_method;
```

Comment:
just return method here
then instead of (method==NULL) in for loop, check it once before the loop